### PR TITLE
Return the exports instead of the instance variable from integrateWasmJS

### PR DESCRIPTION
**Pull request by dschuff**
1) Flatten FFI imports and use V8's pre-standard name 
2) Return the exports instead of the whole instance (supported in V8 and soon in SM)

### CODE REVIEW

**Comment by dschuff** [comment]
I haven't updated wasm.js yet to avoid conflicting with #225

**Comment by kripken** [comment]
We also need something on the other side - the imports in the wast need to match up to the flattened form. We could make a pass for that, or do it in asm2wasm, not sure which is best.

Also, because this needs changes to the wast, I don't think it can be only in the v8 code path. iow, either we flatten for everyone, or for noone, if they all use the same wast.

**Comment by kripken** [comment]
Based on offline discussion it sounds like we don't need to flatten anymore, but I think we do still need the v8 part here to find the right object?

**Comment by dschuff** [comment]
The object name change was added to V8 this morning, so we don't need that. And yes I think @titzer said he was going to do the namespace change too, so maybe we can just close this and I can keep this locally until the namespace change lands.

**Comment by dschuff** [comment]
OK, updated to just the one change that we agreed on now.

File: src/js/wasm.js-post.js [hunk 1]
```diff
@@ -145,7 +145,7 @@ function integrateWasmJS(Module) {
```
Comment (kripken) [inline src/js/wasm.js-post.js]: need `.exports` here too

**Comment by kripken** [comment]
The commits can be squashed into 1, I think?

Also, I don't think we need the `wasm.js` update - if it's just for `.exports`, then when I update the emscripten submodule that will just work, as that code won't be bundled into `wasm.js` anymore.

So basically, lgtm with a single commit of 2 changes that add `.exports`.

**Comment by kripken** [comment]
Sorry, I caused a merge conflict for this. Let me just apply the change myself.

**Pull request closed without merging**